climpred/prediction.py

```diff
@@ -38,19 +47,62 @@ def compute_perfect_model(
         metric (str): `metric` name, see :py:func:`climpred.utils.get_metric_function`.
         comparison (str): `comparison` name, see
                           :py:func:`climpred.utils.get_comparison_function`.
+        dim (str or list): dimension to apply metric over. default: ['member', 'init']
         add_attrs (bool): write climpred compute args to attrs. default: True
 
     Returns:
-        skill (xarray object): skill score.
+        skill (xarray object): skill score with dimension as input without dim.
 
     """
-    supervector_dim = 'svd'
-    metric = get_metric_function(metric, PM_METRICS)
+    if metric in PROBABILISTIC_METRICS:
+        if comparison not in PROBABILISTIC_PM_COMPARISONS:
+            raise ValueError(
+                f'Probabilistic metric {metric} cannot work with comparison \
+                 {comparison}'
+            )
+        stack = False
+        # dim = 'member'
+    elif set(dim) == set(['init', 'member']):
+        supervector_dim = 'svd'
+        stack = True
+    else:
+        if metric in ['pr', 'pearson_r', 'acc']:
+            warnings.warn('ACC doesnt work on dim other than ["init", "member"]')
+        stack = False
+
+    if not stack:
+        supervector_dim = dim
+
     comparison = get_comparison_function(comparison, PM_COMPARISONS)
 
-    forecast, reference = comparison(ds, supervector_dim)
+    forecast, reference = comparison(ds, supervector_dim, stack=stack)
+
+    # in case you want to compute skill over member dim
```

Comment:
Wasn't this just handled with the comparison keyword before?

Reply:
the comparison and metric keywords decide whether stack is `True` or `False`.